src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/MessageHeaderMapper.java

```diff
@@ -0,0 +1,70 @@
+package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
+
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.BooleanType;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.DiagnosticReport;
+import org.hl7.fhir.dstu3.model.Extension;
+import org.hl7.fhir.dstu3.model.MessageHeader;
+import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.StringType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.mesh.RecipientMailboxIdMappings;
+import uk.nhs.digital.nhsconnect.lab.results.model.FhirProfiles;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.utils.ResourceFullUrlGenerator;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+@Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class MessageHeaderMapper {
+    private final UUIDGenerator uuidGenerator;
+    private final ResourceFullUrlGenerator fullUrlGenerator;
+    private final RecipientMailboxIdMappings recipientMailboxIdMappings;
+
+    public MessageHeader mapToMessageHeader(
+            Message message, DiagnosticReport diagnosticReport,
+            Organization performingOrganization, Organization requestingOrganization) {
+        final var messageHeader = new MessageHeader();
```

Comment:
Do we need to set the timestamp? I believe this is mandatory according to the MessageHeader profile